go.mod

```diff
@@ -19,6 +19,7 @@ require (
 	github.com/imdario/mergo v0.3.16
 	github.com/kong/go-apiops v0.1.21
 	github.com/kong/go-kong v0.46.0
+	github.com/kong/kubernetes-ingress-controller/v2 v2.11.1
```

Comment:
This line is effectively adding a module dependency cycle between KIC and decK that we cannot make happen.

Since KIC uses decK as a library, it won't be possible for decK to import KIC as a module. A possible resolution for this would be:
* introduction of an "umbrella" project over the existing decK that would decouple "deck-as-a-library" from "deck-as-a-tool" or
* a massive refactor of deck or KIC.

Reply:
The reason for this dependency is to make use of the API Schema definitions for `KongClusterPlugin`, `KongConsumer`, `KongIngress`, `KongPlugin`, etc. Would it be possible to extract the API Schema definitions into its own module that is imported by both deck and KIC? Similarly to what k8s does here: https://github.com/kubernetes/api

Reply:
Essentially, extract type definitions under `https://github.com/Kong/kubernetes-ingress-controller/tree/main/pkg/apis/configuration` into its own module.

Reply:
The eng/PM decision reached is that we'll unblock `deck2kic`'s dependency on KIC by implementing the proposal tracked by https://github.com/Kong/deck/issues/1060.